ZKUI-190: make sure zenko client is login before calling s3 actions

ChengYanJin opened the pull request:
After we initialize zenko client, login() function is called after the
s3 acitions which casuse the error of
Missing credenticals in config, if using AWS_CONFIG_FILE, set
AWS_SDK_LOAD_CONFIG = 1.

The way we fix it is to add an internal state _isLogin, and check the
state before doing the s3 actions

bert-e commented:
# Hello chengyanjin,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko-ui/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZKUI-190 contains:


* `1.4.2`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `1.5.0`


Please check the `Fix Version/s` of ZKUI-190, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZKUI-190 contains:


* `1.4.2`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `1.4.9`

* `1.5.0`


Please check the `Fix Version/s` of ZKUI-190, or the target
branch of this pull request.

ChengYanJin committed:
ZKUI-190: Make sure the zenko client is login before s3 actions

After we initialize zenko client, login() function is called after the
s3 acitions which casuse the error of
Missing credenticals in config, if using AWS_CONFIG_FILE, set
AWS_SDK_LOAD_CONFIG = 1.

The way we fix it is to add an internal state _isLogin, and check the
state before doing the s3 actions

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/ZKUI-190-assume-role-when-list-buckets` into
`development/1.4`
* [w/1.5/bugfix/ZKUI-190-assume-role-when-list-buckets](https://github.com/scality/zenko-ui/compare/development/1.5...w/1.5/bugfix/ZKUI-190-assume-role-when-list-buckets) will be merged into `development/1.5` 





You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @monpote

* @chengyanjin

* @jbwatenbergscality

ChengYanJin commented:
/approve

ChengYanJin commented:
/status

bert-e commented:
# Status





*Status report is not available.*






*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.4`

* :heavy_check_mark: `development/1.5`




There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.4`

* :heavy_check_mark: `development/1.5`




Please check the status of the associated issue ZKUI-190.

Goodbye chengyanjin.
